src/escape.rs

```diff
@@ -221,10 +252,10 @@ where
                 if let Some(entity) = pat.strip_prefix('#') {
                     let codepoint = parse_number(entity, start..end)?;
                     unescaped.push_str(codepoint.encode_utf8(&mut [0u8; 4]));
-                } else if let Some(value) = named_entity(pat) {
-                    unescaped.push_str(value);
                 } else if let Some(value) = resolve_entity(pat) {
                     unescaped.push_str(value);
+                } else if let Some(value) = named_entity(pat) {
+                    unescaped.push_str(value);
```

Comment:
I also considering to completely remove this `if` then the `unescape_with` will never resolve predefined entities and instead caller will need to call functions for default processing which we make public. Is that something that we want to do?

Reply:
I think it would be poor UX to force the user to declare all predefined entities.  These are built into every other XML parser so I don't see a reason to diverge.

Reply:
Yes, I mean making `named_entity` public (as `resolve_predefined_entity` + `resolve_xml_entity` and `resolve_html5_entity`) so the user can call it after processing it's own entities.